parsec/_parsec/addrs.pyi

```diff
@@ -0,0 +1,258 @@
+    @property
+    def port(self) -> Optional[int]: ...
+    @property
+    def use_ssl(self) -> bool: ...
+    @property
+    def netloc(self) -> str: ...
+    @property
+    def organization_id(self) -> OrganizationID: ...
+    @property
+    def workspace_id(self) -> EntryID: ...
+    @property
+    def encrypted_path(self) -> bytes: ...
+    def get_backend_addr(self) -> BackendAddr: ...
+    def to_url(self) -> str: ...
+    def to_http_redirect_url(self) -> str: ...
+    @classmethod
+    def from_url(
+        cls, url: str, allow_http_redirection: bool = False
+    ) -> BackendOrganizationFileLinkAddr: ...
+    @classmethod
+    def build(
+        cls,
+        organization_addr: BackendOrganizationAddr,
+        workspace_id: EntryID,
+        encrypted_path: bytes,
+    ) -> BackendOrganizationFileLinkAddr: ...
+
+class BackendInvitationAddr:
+    def __init__(
+        self,
+        organization_id: OrganizationID,
+        invitation_type: Any,  # TODO: find correct type
+        token: InvitationToken,
+        hostname: str,
+        port: Optional[int],
+        use_ssl: bool = True,
+    ) -> None: ...
+    def __repr__(self) -> str: ...
+    def __str__(self) -> str: ...
+    def __eq__(self, other: BackendInvitationAddr) -> bool: ...
+    def __ne__(self, other: BackendInvitationAddr) -> bool: ...
+    def __lt__(self, other: BackendInvitationAddr) -> bool: ...
+    def __gt__(self, other: BackendInvitationAddr) -> bool: ...
+    def __le__(self, other: BackendInvitationAddr) -> bool: ...
+    def __ge__(self, other: BackendInvitationAddr) -> bool: ...
+    def __hash__(self) -> int: ...
+    @property
+    def hostname(self) -> str: ...
+    @property
+    def port(self) -> Optional[int]: ...
+    @property
+    def use_ssl(self) -> bool: ...
+    @property
+    def netloc(self) -> str: ...
+    @property
+    def organization_id(self) -> OrganizationID: ...
+    @property
+    def invitation_type(self) -> Any: ...
+    @property
+    def token(self) -> InvitationToken: ...
+    def get_backend_addr(self) -> BackendAddr: ...
+    def to_url(self) -> str: ...
+    def to_http_redirect_url(self) -> str: ...
+    def generate_organization_addr(self, root_verify_key: VerifyKey) -> BackendOrganizationAddr: ...
+    @classmethod
+    def from_url(
+        cls, url: str, allow_http_redirection: bool = False
+    ) -> BackendOrganizationFileLinkAddr: ...
+    @classmethod
+    def build(
+        cls,
+        organization_addr: BackendOrganizationAddr,
+        workspace_id: EntryID,
+        encrypted_path: bytes,
+    ) -> BackendOrganizationFileLinkAddr: ...
+
+class BackendPkiEnrollmentAddr:
+    def __init__(
+        self,
+        organization_id: OrganizationID,
+        hostname: str,
+        port: Optional[int],
+        use_ssl: bool = True,
+    ) -> None: ...
+    def __repr__(self) -> str: ...
+    def __str__(self) -> str: ...
+    def __eq__(self, other: BackendInvitationAddr) -> bool: ...
```

Comment:
Can't we compare with BackendPkiEnrollmentAddr ? Is the binding missing too ?

Reply:
That's a typo

Reply:
done